textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcherTest.java

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * @author Parag Saraogi
+ *
+ */
+
+public class FuzzyTokenMatcherTest {
+
+    private FuzzyTokenMatcher fuzzyMatcher;
+    private IDataWriter dataWriter;
+    private DataStore dataStore;
+    private Analyzer analyzer;
+    
+    @Before
+    public void setUp() throws Exception {
+        dataStore = new DataStore(DataConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+        analyzer = new StandardAnalyzer();
+        dataWriter = new DataWriter(dataStore, analyzer);
+        dataWriter.clearData();
+        dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+    }
+    
+    @After
+    public void cleanUp() throws Exception {
+        dataWriter.clearData();
+    }
+    
+    public List<ITuple> getResults(String query,double threshold, ArrayList<Attribute> attributeList, boolean getSpan) throws DataFlowException, ParseException {
+
+        IPredicate predicate = new FuzzyTokenPredicate(query, attributeList, analyzer, dataStore, threshold, getSpan);
+        fuzzyMatcher = new FuzzyTokenMatcher(predicate);
+        fuzzyMatcher.open();
+
+        List<ITuple> results = new ArrayList<>();
+        ITuple nextTuple = null;
+        while ((nextTuple = fuzzyMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+
+        return results;
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithNoResults() throws Exception {
+        String query = "Twelve Angry Men Cafe";
+        double threshold = 0.5; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+        List<ITuple> results = getResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherForStringField() throws Exception {
+        String query = "tom hanks";
+        double threshold = 1; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.FIRST_NAME_ATTR);
+        List<ITuple> results = getResults(query, threshold, attributeList, getSpan);
+        //Assert.assertEquals(0,results.size());
+    }
+    
+    @Test
+    public void FuzzyTokenMatcherWithoutSpan() throws Exception {
+        String query = "Twelve Angry Men";
+        double threshold = 0.5; 	//The ratio of tokens that need to be matched
+        boolean getSpan = false;
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+        
+        Attribute[] schemaAttributes = new Attribute[TestConstants.ATTRIBUTES_PEOPLE.length];
+        for(int count = 0; count < schemaAttributes.length; count++) {
+            schemaAttributes[count] = TestConstants.ATTRIBUTES_PEOPLE[count];
+        }
```

Comment:
Why copying the attribute array? Could we just use TestConstants.ATTRIBUTES_PEOPLE?